networkx/algorithms/euler.py

```diff
@@ -147,3 +158,203 @@ def eulerian_circuit(G, source=None):
             arbitrary_edge = next(edges(current_vertex))
             vertex_stack.append(get_vertex(arbitrary_edge))
             g.remove_edge(*arbitrary_edge)
+
+
+def is_semieulerian(G):
+    """Returns ``True`` if and only if ``G`` has an Eulerian path.
+
+    An Eulerian path is a path that crosses every edge in G
+    exactly once.
+
+    Parameters
+    ----------
+    G: NetworkX Graph, DiGraph, MultiGraph or MultiDiGraph
+        A directed or undirected Graph or MultiGraph.
+
+    Returns
+    -------
+    True, False
+
+    See Also
+    --------
+    is_eulerian()
+
+    Examples
+    --------
+    >>> G = nx.DiGraph([(1,2),(2,3)])
+    >>> nx.is_semieulerian(G)
+    True
+    """
+    is_directed = G.is_directed()
+
+    # Verify that graph is connected, short circuit
+    if is_directed and not nx.is_weakly_connected(G):
+        return False
+
+    # is undirected
+    if not is_directed and not nx.is_connected(G):
+        return False
+
+    # Not all vertex have even degree, check if exactly two vertex
+    # have odd degrees.  If yes, then there is an Euler path. If not,
+    # raise an error (no Euler path can be found)
+
+    # if the odd condition is not meet, raise an error.
+    start = _find_path_start(G)
+    if not start:
```

Comment:
The `_find_path_start` function states that it returns ``None`` if there is no valid start node. If that is correct, then this and the next two lines should be `return start is not None`.